backend/battle/forms.py

```diff
@@ -36,41 +44,35 @@ class Meta:
             "pokemon_3",
         ]
 
-    pokemon_1 = forms.IntegerField(
-        label="Pokemon 1",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-    pokemon_2 = forms.IntegerField(
-        label="Pokemon 2",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-    pokemon_3 = forms.IntegerField(
-        label="Pokemon 3",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-
-    def __init__(self, *args, **kwargs):
-        super(TeamForm, self).__init__(*args, **kwargs)
-        self.fields['battle'].widget = forms.HiddenInput()
-        self.fields['trainer'].widget = forms.HiddenInput()
+    battle = forms.ModelChoiceField(queryset=Battle.objects.all())
+    trainer = forms.ModelChoiceField(queryset=User.objects.all())
+    pokemon_1 = forms.CharField(widget=forms.Textarea)
+    pokemon_2 = forms.CharField(widget=forms.Textarea)
+    pokemon_3 = forms.CharField(widget=forms.Textarea)
 
     def clean(self):
         cleaned_data = super().clean()
+        pokemon_1 = self.cleaned_data.get('pokemon_1')
+        pokemon_2 = self.cleaned_data.get('pokemon_2')
+        pokemon_3 = self.cleaned_data.get('pokemon_3')
+
+        if not pokemon_1 or not pokemon_2 or not pokemon_3:
+            raise forms.ValidationError('ERROR: Select all pokemons')
+
+        for pokemon in [pokemon_1, pokemon_2, pokemon_3]:
+            response = requests.get(urljoin(settings.POKE_API_URL, pokemon))
+            if response.status_code == 404:
+                raise forms.ValidationError('ERROR: Type the correct pokemon name ')
```

Comment:
This `for` and this response only checks if the pokemon exist in the API, right?
So I think this should be moved to a service function and you should call the service here.

The service can return `False` if any pokemon doesn't exist, and in that case you can raise this exception.

Reply:
Great, I'll make this improvement.